flatdata-py/flatdata/archive.py

```diff
@@ -79,24 +79,23 @@ def open(cls, storage, name, initializer, is_optional=False):
         return initializer(nested_storage)
 
     def size_in_bytes(self):
-        return sum(
-            self.__getattr__(resource).size_in_bytes()
-            for resource in self._RESOURCES.keys()
-        )
+        return sum( resource_value.size_in_bytes() for resource_value in  ( self.__getattr__(resource)
+            for resource in self._RESOURCES.keys() if resource ) if resource_value )
 
     def __len__(self):
         return len(self._RESOURCES)
 
     def _open_resource(self, name):
         resource_signature = self._RESOURCES[name]
-        self._check_non_subarchive_schema(name, resource_signature)
-        resource = resource_signature.container.open(storage=self._resource_storage,
-                                                     name=name,
-                                                     initializer=resource_signature.initializer,
-                                                     is_optional=resource_signature.is_optional)
+        if self._check_non_subarchive_schema(name, resource_signature):
```

Comment:
This condition will make `resource` on line 95 unassigned.

Reply:
Done. Missed that during merge.